### PR TITLE
fix in-perfect patch for potential goroutine leak

### PR DESCRIPTION
**Pull request by charlesxsh**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
The original patch #102635 might not be suitable for all situations.

#### Which issue(s) this PR fixes:
fix in-perfect patch for potential goroutine leak

Fixes #102635

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by system-pclub** [comment]
In total, the send operation is conducted len(tc.activePods) times. Thus, increasing the buffer size to len(tc.activePods) is more reasonable.

File: pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func TestManager(t *testing.T) {
 				GracePeriod int64
 			}
 
-			podKillChan := make(chan PodKillInfo, 1)
+			podKillChan := make(chan PodKillInfo, len(tc.activePods))
```
Comment (matthyx) [inline pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux_test.go:217]: why was it 1 before?

**Pull request closed without merging**